internal/contract/controlplane.go

```diff
@@ -351,3 +353,56 @@ func (c *ControlPlaneMachineTemplate) NodeDeletionTimeout() *Duration {
 		path: Path{"spec", "machineTemplate", "nodeDeletionTimeout"},
 	}
 }
+
+// ReadinessGates provides access to control plane's ReadinessGates.
+func (c *ControlPlaneContract) ReadinessGates() *ReadinessGates {
+	return &ReadinessGates{}
+}
+
+// ReadinessGates provides a helper struct for working with ReadinessGates.
+type ReadinessGates struct{}
+
+// Path returns the path of the ReadinessGates.
+func (m *ReadinessGates) Path() Path {
+	return Path{"spec", "readinessGates"}
+}
+
+// Get gets the ReadinessGates object.
+func (m *ReadinessGates) Get(obj *unstructured.Unstructured) ([]clusterv1.MachineReadinessGate, error) {
+	unstructuredValue, ok, err := unstructured.NestedSlice(obj.UnstructuredContent(), m.Path()...)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve control plane metadata.labels")
+	}
+	if !ok {
+		return nil, nil
+	}
+	jsonValue, err := json.Marshal(&unstructuredValue)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to Marshal control plane spec.readinessGates")
+	}
+	var readinessGates []clusterv1.MachineReadinessGate
+	if err := json.Unmarshal(jsonValue, &readinessGates); err != nil {
+		return nil, errors.Wrap(err, "failed to Unmarshal control plane spec.readinessGates")
+	}
+	return readinessGates, nil
+}
+
+// Set sets the ReadinessGates value.
+func (m *ReadinessGates) Set(obj *unstructured.Unstructured, readinessGates []clusterv1.MachineReadinessGate) error {
+	if readinessGates == nil {
```

Comment:
My expectation would be that we set the field to nil or remove it, not that we do nothing